h2o-core/src/main/java/water/H2O.java

```diff
@@ -2197,21 +2205,52 @@ public static void gc() {
     }
   }
 
+  private static boolean JAVA_CHECK_PASSED = false;
+
   /**
    * Check if the Java version is not supported
    *
    * @return true if not supported
    */
-  public static boolean checkUnsupportedJava() {
+  public static boolean checkUnsupportedJava(String[] args) {
+    if (JAVA_CHECK_PASSED)
+      return false;
     if (Boolean.getBoolean(H2O.OptArgs.SYSTEM_PROP_PREFIX + "debug.noJavaVersionCheck")) {
       return false;
+    } 
+    boolean unsupported = runCheckUnsupportedJava(args);
+    if (!unsupported) {
+      JAVA_CHECK_PASSED = true;
     }
+    return unsupported;
+  }
 
+  static boolean runCheckUnsupportedJava(String[] args) {
     if (!JavaVersionSupport.runningOnSupportedVersion()) {
-      System.err.println(String.format("Only Java versions %d-%d are supported, system version is %s",
+      Throwable error = null;
+      boolean allowUnsupported = ArrayUtils.contains(args, "-allow_unsupported_java");
+      if (allowUnsupported) {
+        boolean checkPassed = false;
+        try {
+          checkPassed = dynamicallyInvokeJavaSelfCheck();
+        } catch (Throwable t) {
+          error = t;
+        }
+        if (checkPassed) {
+          Log.warn("H2O is running on a version of Java (" + System.getProperty("java.version") + ") that was not certified at the time of the H2O release. " + 
+                  "For production use please use a certified Java version.");
+          return false;
+        }
```

Comment:
We can be more transparent here and tell the user that we know that he forces his java version, but H2O cannot start.

Reply:
@valenad1 I am not sure what you mean here - can you please provide more details?

Reply:
I meant this

> if user use unsupported java and don't pass your check -> tell him that he must use different java because allow_unsupported_java don't work

Reply:
Will do, thank you!